src/agent0/chainsync/analysis/db_to_analysis.py

```diff
@@ -0,0 +1,123 @@
+"""Functions to gather data from postgres, do analysis, and add back into postgres"""
+
+from __future__ import annotations
+
+import pandas as pd
+from sqlalchemy.orm import Session
+
+from agent0.chainsync.db.hyperdrive import (
+    PositionSnapshot,
+    get_checkpoint_info,
+    get_current_positions,
+    get_latest_block_number_from_positions_snapshot_table,
+)
+from agent0.chainsync.df_to_db import df_to_db
+from agent0.ethpy.hyperdrive import HyperdriveReadInterface
+
+from .calc_position_value import calc_closeout_value
+
+pd.set_option("display.max_columns", None)
+
+
+def db_to_analysis(
+    db_session: Session,
+    interfaces: list[HyperdriveReadInterface],
+    calc_pnl: bool = True,
+) -> None:
+    """Function to query postgres data tables and insert to analysis tables.
+    Executes analysis on a batch of blocks, defined by start and end block.
+
+    Arguments
+    ---------
+    db_session: Session
+        The initialized db session.
+    interfaces: list[HyperdriveReadInterface] | None, optional
+        A collection of Hyperdrive interface objects, each connected to a pool.
+    calc_pnl: bool
+        Whether to calculate pnl. Defaults to True.
+    """
+
+    # Snapshot wallet to table.
+    # This function takes care of not adding duplicate entries.
+    # TODO there may be time and memory concerns here if we're spinning up from
+    # scratch and there's lots of trades/pools.
+    snapshot_positions_to_db(
+        interfaces,
+        wallet_addr=None,
+        calc_pnl=calc_pnl,
+        db_session=db_session,
+    )
+
+
+def snapshot_positions_to_db(
+    interfaces: list[HyperdriveReadInterface], wallet_addr: str | None, calc_pnl: bool, db_session: Session
+):
+    """Function to query the trade events table and takes a snapshot
+    of the current positions and pnl.
+
+    ..note:: This function does not scale well in simulation mode, as this table grows
```

Comment:
I think for sphinx to format correctly you want the note to be indented wrt the `..note::` line

Reply:
you also want a space between `..` and `note::`